src/main/java/com/minecolonies/coremod/colony/Structures.java

```diff
@@ -60,200 +117,863 @@ public static void init()
 
-            if ("jar".equals(uri.getScheme()))
-            {
-                try (FileSystem fileSystem = FileSystems.newFileSystem(uri, Collections.emptyMap()))
-                {
-                    basePath = fileSystem.getPath(SCHEMATICS_ASSET_PATH);
-                    loadStyleMaps(basePath);
-                }
-            }
-            else
-            {
-                basePath = Paths.get(uri);
-                loadStyleMaps(basePath);
-            }
+        final File schematicsFolder = MineColonies.proxy.getSchematicsFolder();
+        if (schematicsFolder != null)
+        {
+            Log.getLogger().info("Load additionnal huts or decorations from " + schematicsFolder + SCHEMATICS_SEPARATOR + SCHEMATICS_PREFIX);
+            checkDirectory(schematicsFolder.toPath().resolve(SCHEMATICS_PREFIX).toFile());
+            loadSchematicsForPrefix(schematicsFolder.toPath(), SCHEMATICS_PREFIX);
+        }
 
-            final File decorationFolder;
+        final File cacheSchematicFolder = Structure.getCachedSchematicsFolder();
+        if (cacheSchematicFolder != null)
+        {
+            checkDirectory(cacheSchematicFolder);
+            Log.getLogger().info("Load cached schematic from " + cacheSchematicFolder + SCHEMATICS_SEPARATOR + SCHEMATICS_CACHE);
+            checkDirectory(cacheSchematicFolder.toPath().resolve(SCHEMATICS_CACHE).toFile());
+            loadSchematicsForPrefix(cacheSchematicFolder.toPath(), SCHEMATICS_CACHE);
+        }
 
-            if (FMLCommonHandler.instance().getMinecraftServerInstance() == null)
-            {
-                decorationFolder = new File(Minecraft.getMinecraft().mcDataDir, "minecolonies/decorations");
-            }
-            else
+        if (md5Map.size() == 0)
+        {
+            Log.getLogger().error("Error loading StructureProxy directory. Things will break!");
+        }
+    }
+
+    /**
+     * load the schematics from the jar.
+     */
+    private static void loadStyleMapsJar()
+    {
+        URI uri = null;
+        try
+        {
+            uri = ColonyManager.class.getResource(SCHEMATICS_ASSET_PATH).toURI();
+        }
+        catch (@NotNull URISyntaxException e)
+        {
+            Log.getLogger().error("loadStyleMaps : ",e);
+            return;
+        }
+
+        if ("jar".equals(uri.getScheme()))
+        {
+            try (FileSystem fileSystem = FileSystems.getFileSystem(uri))
             {
-                decorationFolder = new File(FMLCommonHandler.instance().getMinecraftServerInstance().getDataDirectory(), "minecolonies/decorations");
+                final Path basePath = fileSystem.getPath(SCHEMATICS_ASSET_PATH);
+                Log.getLogger().info("Load huts or decorations from jar");
+                loadSchematicsForPrefix(basePath, SCHEMATICS_PREFIX);
             }
-
-            if (!decorationFolder.exists() && !decorationFolder.mkdirs())
+            catch (@NotNull IOException | FileSystemNotFoundException e1)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1166)